package-builder/build-manifest.nix

```diff
@@ -111,6 +111,11 @@ in
       ''
     )
     + ''
+      # Check that the build populated $out.
+      if [ ! -e "$out" ]; then
+        echo 'ERROR: build did not produce expected $out' 1>&2
```

Comment:
See comment below about making error message more consistent with Nix, because in the pure build mode the Nix error will happen first.

Reply:
With the implementation currently in this PR we don't get the Nix error `failed to produce output path for output 'out'` because our new check bails early:
```
…
foo> bar
foo> ERROR: build did not produce expected $out
error: builder for '/nix/store/b9lmcvq3f6ra86wg3dmbahwhaqqx4r0j-foo-0.0.0.drv' failed with exit code 1;
       last 10 log lines:
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/ilaf1w22bxi6jsi45alhmvvdgy4ly3zs-patch-shebangs.sh
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/cickvswrvann041nqxb0rxilc46svw1n-prune-libtool-files.sh
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/xyff06pkhki3qy1ls77w10s0v79c9il0-reproducible-builds.sh
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/ngg1cv31c8c7bcm2n8ww4g06nq7s4zhm-set-source-date-epoch-to-latest.sh
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/gps9qrh99j7g02840wv5x78ykmz30byp-strip.sh
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/814xig7dadb0pf665dh2dw3fnpwcwz20-environment
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/04w1mgkhawycg64s4q3n8mkcvma5ppx0-apple-framework-CoreFoundation-11.0.0
       > calling 'envHostTargetHook' function hook 'sdkRootHook' /nix/store/lybzq28v1ay0zac24hp0pw38s76ijvsj-libobjc-11.0.0
       > foo
       > ERROR: build did not produce expected $out
       For full logs, run 'nix log /nix/store/b9lmcvq3f6ra86wg3dmbahwhaqqx4r0j-foo-0.0.0.drv'.
make: *** [/nix/store/xrcbbjwbs2zzsbrz3nxm753dsa4ivxzx-package-builder-1.0.0/libexec/flox-build.mk:321: foo_nix_sandbox_build] Error 1
rm /var/folders/8q/spckhr654cv4xrcv0fxsrlvc0000gn/T/nix-shell.h7ZVNx/291fe2a7-foo-build.bash
make: Leaving directory '/Users/dcarley/tmp'
❌ ERROR: Build failed with status: exit status: 2
```

(where "foo" is the name of the build and "bar" is the output of the build command)

Reply:
OK ... sounds good.  👍 

So the observations boil down to this:
* as long as we can demonstrate consistent flox-supplied error messages and hints for both sandbox modes then we're good
* it may be useful to provide the _value_ of `$out` in a hint because that path is in fact stable across `sandbox=off` builds

Separately, where did those "calling ... function hook" messages come from?!? Those are new in the last few weeks/months, possibly on account of a nixpkgs upgrade upstream? We need to squash those messages coming out of the pure build mode.

Reply:
> Separately, where did those "calling ... function hook" messages come from?!? Those are new in the last few weeks/months, possibly on account of a nixpkgs upgrade upstream? We need to squash those messages coming out of the pure build mode.

Ticketed in https://github.com/flox/flox/issues/2127